src/tuned_models.jl

```diff
@@ -198,10 +239,41 @@ function TunedModel(; model=nothing,
                     check_measure=true,
                     cache=true)
 
-    range === nothing && error("You need to specify `range=...`.")
-    model == nothing && error("You need to specify model=... .\n"*
-                              "If `tuning=Explicit()`, any model in the "*
-                              "range will do. ")
+    # either `models` is specified and `tuning` is set to `Explicit`,
+    # or `models` is unspecified and tuning will fallback to `Grid()`
+    # unless it is itself specified:
+    if models !== nothing
+        if tuning === nothing
+            tuning = Explicit()
+        elseif !(tuning isa Explicit)
+            throw(ERR_NEED_EXPLICIT)
+        end
+    else
+        tuning === nothing && (tuning = Grid())
+    end
+
+    # either a `model` is specified or we are in the case
+    # `tuning=Explicit()`. In the latter case, either `models` or
+    # `range` stores the iterator, but ultimately `range` must do so:
+    if model === nothing
+        if tuning isa Explicit
+            if models === nothing
+                range === nothing && throw(ERR_SPECIFY_RANGE_OR_MODELS)
+            else
+                range = models
+            end
+        else
+            throw(ERR_SPECIFY_MODEL)
+        end
+    else
+        models === nothing || throw(ERR_BOTH_DISALLOWED)
+        range === nothing && throw(ERR_SPECIFY_RANGE)
```

Comment:
These combinations of `A || B` and `A && B` where `B` has some side-effect are super super confusing for me to read. I know that it is somewhat idiomatic Julia (it is in Julia base), but I think that it would be more readable by sticking to `&&` where possible.
 
```
model !== nothing && B()
```
instead of 
```
model === nothing || B()
```

But, yeah, this is my opinion (and probably of new(ish) Julia programmers), but you can ignore my comment, of course, if you want

Reply:
Yeah, fair enough comment, and I do appreciate the feedback. I've actually come to find it easier to read this way.  I read `A || command` as "Assuming `A` is true, you can proceed to the next thing, otherwise do `command`.  It "avoids" a negative and when there are many of these it can be confusing (to me).

I've fixed it here but, it's really all over the place in MLJ code and elsewhere.

Reply:
After your "Assuming `A` is true, you can proceed otherwise do `command`" , I find it easier to read now. It's something that I have to get used to in Julia I'm afraid. Thanks for being open about it